openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -23,94 +120,158 @@
 
 
 logger = logging.getLogger(__name__)
-logging.basicConfig(
-    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
-)
-
-# The 10000 is a bit arbitrary, but needs to be larger than the mean
-# number of uses per file (globally) in the response_json, or we will
-# fail without a continuation token.  The largest example seen so far
-# had a little over 1000 uses
-MEAN_GLOBAL_USAGE_LIMIT = 10000
-HOST = "commons.wikimedia.org"
-PAGES_PATH = ["query", "pages"]
-
-IMAGE_MEDIATYPES = {"BITMAP", "DRAWING"}
-AUDIO_MEDIATYPES = {"AUDIO"}
-# Other types available in the API are OFFICE for pdfs and VIDEO
 
 
 class WikimediaCommonsDataIngester(ProviderDataIngester):
     providers = {
         "image": prov.WIKIMEDIA_DEFAULT_PROVIDER,
         "audio": prov.WIKIMEDIA_AUDIO_PROVIDER,
     }
-    endpoint = f"https://{HOST}/w/api.php"
+    host = "commons.wikimedia.org"
+    endpoint = f"https://{host}/w/api.php"
     headers = {"User-Agent": prov.UA_STRING}
 
+    # The 10000 is a bit arbitrary, but needs to be larger than the mean
+    # number of uses per file (globally) in the response_json, or we will
+    # fail without a continuation token.  The largest example seen so far
+    # had a little over 1000 uses
+    mean_global_usage_limit = 10000
+    # Total number of attempts to retrieve global usage pages for a given batch.
+    # These can be very large, so we need to limit the number of attempts otherwise
+    # the DAG will hit the timeout when it comes across these items.
+    # See: https://github.com/WordPress/openverse-catalog/issues/725
+    max_page_iteration_before_give_up = 100
+
+    image_mediatypes = {"BITMAP", "DRAWING"}
+    audio_mediatypes = {"AUDIO"}
+    # Other types available in the API are OFFICE for pdfs and VIDEO
+
     # The batch_limit applies to the number of pages received by the API, rather
     # than the number of individual records. This means that for Wikimedia Commons
     # setting a global `INGESTION_LIMIT` will still limit the number of records, but
     # the exact limit may not be respected.
     batch_limit = 250
 
+    class ReturnProps:
+        """Different sets of properties to return from the API."""
+
+        # All normal media info, plus popularity info by global usage
+        query_all = "imageinfo|globalusage"
+        # Just media info, used where there's too much global usage data to parse
+        query_no_popularity = "imageinfo"
+
+        # All media info we care about
+        media_all = "url|user|dimensions|extmetadata|mediatype|size|metadata"
+        # Everything minus the metadata, which is only necessary for audio and can
+        # balloon for PDFs which are considered images by Wikimedia
+        media_no_metadata = "url|user|dimensions|extmetadata|mediatype|size"
+
+    # MappingProxy effectively provides a frozen dictionary, since this is defined
+    # at the class level and technically mutable.
+    default_props = MappingProxyType(
```

Comment:
TIL about `MappingProxyType`!

Reply:
Same! I would have used [frozendict](https://pypi.org/project/frozendict/) but I didn't want to introduce a new library; I was glad to find that this was already available 🙂